lib/convertor.js

```diff
@@ -87,37 +105,58 @@ class MongoToKnex {
     }
 
     /**
-     * @TODO: This implementation serves currently only one use case:
+     * @TODO: This implementation serves currently the following use cases:
      *
      * - OR conjunctions for many-to-many relations
+     * - filter by multiple relation columns
+     * - filter by columns on the join table
      */
-    buildRelationQuery(qb, relation) {
+    buildRelationQuery(qb, relations) {
         debug(`(buildRelationQuery)`);
 
         if (debugExtended.enabled) {
-            debugExtended(`(buildRelationQuery) ${JSON.stringify(relation)}`);
+            debugExtended(`(buildRelationQuery) ${JSON.stringify(relations)}`);
         }
 
-        if (relation.config.type === 'manyToMany') {
-            if (isCompOp(relation.operator)) {
-                const comp = compOps[relation.operator] || '=';
-
-                // CASE: post.id IN (SELECT ...)
-                qb.where(`${this.tableName}.id`, 'IN', function () {
-                    return this
-                        .select(`${relation.config.join_table}.${relation.config.join_from}`)
-                        .from(`${relation.config.join_table}`)
-                        .innerJoin(`${relation.config.tableName}`, `${relation.config.tableName}.id`, '=', `${relation.config.join_table}.${relation.config.join_to}`)
-                        .where(`${relation.config.tableName}.${relation.column}`, comp, relation.value);
-                });
-            } else {
-                debug('unknown operator');
-            }
+        const groupedRelations = _.groupBy(relations, 'table');
 
-            return;
+        if (debugExtended.enabled) {
+            debugExtended(`(buildRelationQuery) grouped: ${JSON.stringify(relations)}`);
         }
 
-        debug('not implemented');
+        // CASE: {tags: [where clause, where clause], authors: [where clause, where clause]}
+        _.each(Object.keys(groupedRelations), (key) => {
+            debug(`(buildRelationQuery) build relation for ${key}`);
+
+            const statements = groupedRelations[key];
+
+            // CASE: any statement for the same relation should contain the same config
```

Comment:
This also implies that operators are also of the same type (so they pass `isCompOp(reference.operator)` condition in the same way), right?

Reply:
I just fulfilled the test cases which were added. If the operator is different, we might need to tweak the code, because we operate on the reference statement currently.

This code is for sure not optimal, but it does not make sense to refactor already, see #17.

Reply:
:+1:

Reply:
All existing cases should be covered in the next two weeks 👍 

For now, we 

- try to get rid of potential blockers
- hard coded queries/logic in the filter plugin e.g. filter by join table column

To be able to finish up the first version of https://github.com/TryGhost/Ghost/pull/10159 by tomorrow.